azure-quantum/azure/quantum/qiskit/job.py

```diff
@@ -95,7 +95,8 @@ def result(self, timeout=None, sampler_seed=None):
 
         result_type = Result
         if self._azure_job.details.output_data_format == RESOURCE_ESTIMATOR_OUTPUT_DATA_FORMAT:
-            result_type = ResourceEstimatorResult
+            is_simple_job = not isinstance(results['data'], list)

```

Comment:
Do you know if data is guaranteed to be on the result object? Will it still come back as empty array if job is not successful? What if it fails validation at data plane before reaching resource estimator?

Reply:
Yes, `data` is guaranteed to be on the `result` object and is an empty object in case of an error. However, I added an additional check to only check for batching jobs on successful jobs.